site/oauth2.md

```diff
@@ -0,0 +1,464 @@
+| `auth_oauth2.https.hostname_verification`| Enable wildcard-aware hostname verification for key server. Available values: `wildcard`, `none`. Default is `none`.
+| `auth_oauth2.algorithms`                 | Restrict [the usable algorithms](https://github.com/potatosalad/erlang-jose#algorithm-support).
+| `auth_oauth2.verify_aud`                 | [Verify token's `aud`](#token-validation).
+
+For example:
+
+Configure with key files
+<pre class="lang-ini">
+auth_oauth2.resource_server_id = new_resource_server_id
+auth_oauth2.additional_scopes_key = my_custom_scope_key
+auth_oauth2.default_key = id1
+auth_oauth2.signing_keys.id1 = test/config_schema_SUITE_data/certs/key.pem
+auth_oauth2.signing_keys.id2 = test/config_schema_SUITE_data/certs/cert.pem
+auth_oauth2.algorithms.1 = HS256
+auth_oauth2.algorithms.2 = RS256
+</pre>
+Configure with key server
+<pre class="lang-ini">
+auth_oauth2.resource_server_id = new_resource_server_id
+auth_oauth2.jwks_url = https://my-jwt-issuer/jwks.json
+auth_oauth2.https.cacertfile = test/config_schema_SUITE_data/certs/cacert.pem
+auth_oauth2.https.peer_verification = verify_peer
+auth_oauth2.https.depth = 5
+auth_oauth2.https.fail_if_no_peer_cert = true
+auth_oauth2.https.hostname_verification = wildcard
+auth_oauth2.algorithms.1 = HS256
+auth_oauth2.algorithms.2 = RS256
+</pre>
+### <a id="resource-server-id" class="anchor" href="#resource-server-id">Resource Server ID and Scope Prefixes</a>
+
+OAuth 2.0 (and thus UAA-provided) tokens use scopes to communicate what set of permissions particular
+client has been granted. The scopes are free form strings.
+
+`resource_server_id` is a prefix used for scopes in UAA to avoid scope collisions (or unintended overlap).
+It is an empty string by default.
+
+### <a id="token-validation" class="anchor" href="#token-validation">Token validation</a>
+
+When RabbitMQ receives a JWT token, it validates it before accepting it.
+
+#### Must be digitally signed
+
+The token must carry a digital signature and optionally a `kid` header attribute which identifies the key RabbitMQ should
+use to validate the signature.
+
+#### Must not be expired
+
+RabbitMQ uses this field `exp` ([exp](https://tools.ietf.org/html/rfc7519#page-9)) to validate the token if present.
+It contains the expiration time after which the JWT MUST NOT be accepted for processing.
+
+#### Audience must have/match the resource_server_id
+
+The `aud` ([Audience](https://tools.ietf.org/html/rfc7519#page-9)) identifies the recipients and/or resource_server of the JWT. By default, **RabbitMQ uses this field to validate the token** although we can disable it by setting `verify_aud` to `false`.  When it set to `true`, this attribute must either match the `resource_server_id` setting or in case of a list, it must contain the `resource_server_id`.
```

Comment:
The `aud` ([Audience](https://tools.ietf.org/html/rfc7519#page-9)) field identifies the recipients and/or resource_server of the JWT. By default, **RabbitMQ uses this field to validate the token**, although this can be disabled by setting `verify_aud` to `false`. When set to `true`, this attribute must either match the `resource_server_id` setting or, in case of a list, it must contain the `resource_server_id`.